tools/new_minor_OCP_version.py

```diff
@@ -120,6 +136,109 @@ def cmd_with_git_ssh_key(key_file):
+    if task is None:
+        logging.info("Not creating PR because ticket already exists")
+        return
+
+    clone_assisted_service(args.github_user_password)
+
+    change_version_in_files(rhcos_future_default_release, rhcos_future_latest_release, RCHOS_RELEASE_REPLACE_CONTEXT)
+
+    rchos_version_from_iso = get_rchos_version_from_iso(rhcos_future_latest_release)
+    rhcos_version_from_default = release_json[future_version]['rhcos_version']
+
+    change_version_in_files(rhcos_version_from_default, rchos_version_from_iso, RCHOS_VERSION_REPLACE_CONTEXT)
+    cmd(["make", "update-ocp-version"], cwd=ASSISTED_SERVICE_CLONE_DIR)
+    branch = commit_and_push_version_update_changes(rhcos_future_latest_release, task)
+
+    github_pr = open_pr(args, rhcos_future_default_release, rhcos_future_latest_release, task)
+    unhold_pr(github_pr)
+
+
+def rhcos_version_update(args):
+    logger.info("Searching and updating RHCOS version")
+    latest_ocp_version = version.parse(get_latest_ocp_version())
+
+    ocp_version_major = "{major}.{minor}".format(major=latest_ocp_version.major, minor=latest_ocp_version.minor,
+                                                 micro=latest_ocp_version.micro)
+    release_json = get_default_release_json()
+    rhcos_default_release = get_rchos_default_release(ocp_version_major, release_json)
+    rhcos_latest_release = get_rchos_latest_release(RCHOS_LATEST_RELEASE_URL.format(version=ocp_version_major))
+
+    if rhcos_latest_release == rhcos_default_release:
+        logging.info(f"RCHOS version {rhcos_latest_release} is up to date")
+        return
+
+    jira_client, task = create_task(args, RCHOS_TICKET_DESCRIPTION, rhcos_default_release, rhcos_latest_release)
+    if task is None:
+        logging.info("Not creating PR because ticket already exists")
+        return
+
+    clone_assisted_service(args.github_user_password)
+
+    change_version_in_files(rhcos_default_release, rhcos_latest_release, RCHOS_RELEASE_REPLACE_CONTEXT)
+
+    rchos_version_from_iso = get_rchos_version_from_iso(rhcos_latest_release)
+    rhcos_version_from_default = release_json[ocp_version_major]['rhcos_version']
+
+    change_version_in_files(rhcos_version_from_default, rchos_version_from_iso, RCHOS_VERSION_REPLACE_CONTEXT)
+    cmd(["make", "update-ocp-version"], cwd=ASSISTED_SERVICE_CLONE_DIR)
+    branch = commit_and_push_version_update_changes(rhcos_latest_release, task)
+
+    github_pr = open_pr(args, rhcos_default_release, rhcos_latest_release, task)
+    unhold_pr(github_pr)
+
+def get_rchos_version_from_iso(rhcos_latest_release):
+    live_iso_url = RCHOS_LATEST_LIVE_ISO_URL.format(version=rhcos_latest_release)
+    with tempfile.NamedTemporaryFile() as tmp_live_iso_file:
+        subprocess.check_output(
+            DOWNLOAD_LIVE_ISO_CMD.format(live_iso_url=live_iso_url, out_file=tmp_live_iso_file.name), shell=True)
+        try:
+            os.remove("tmp/zipl.prm")
+        except:
```

Comment:
Don't use broad except, at-the very least use `except Exception` to avoid capturing exceptions caused by kill / interrupt signals that might come from a user trying to stop the job